### ISSUE TITLE
Hello all, I need a help. I want use my own ssl certificate. Is it possible ? If yes, how can I do that ?

### ISSUE THREAD

**Issue by fahimistiaq91** [goharbor/harbor-helm#1521]

**Comment by aneurinprice** [comment]
For the web interface ingress or the internal tls?

**Comment by fahimistiaq91** [comment]
For the web interface

**Comment by fahimistiaq91** [comment]
my expose type is load balance and i replace tls.crt & tls.key file with own in cert folder. change values.yaml field certSource from auto to secret. Generate secret with my own crt and key file and put the secret name in values.yaml file.

**Comment by aneurinprice** [comment]
Typically you would want to configure this in your ingress controller

**Comment by fahimistiaq91** [comment]
yes

**Comment by aneurinprice** [comment]
The usual setup is to use an ingress controller (nginx) along with cert-manager. From your description however, it sounds like you arent actually creating the secret needed. Can you confirm that you are creating the secret that you are sourcing?

**Comment by fahimistiaq91** [comment]
Yes, I created the secret with my own crt and key file with this command "kubectl create secret tls harbor-cert-secret --cert=/usr/local/bin/harbor/cert/tls.crt --key=/usr/local/bin/harbor/cert/tls.key"

**Comment by aneurinprice** [comment]
I see. I am going to assume they are both in the same namespace?

What is the actual problem you are seeing? Do you have any logs for me to look at?

**Comment by fahimistiaq91** [comment]
The problem is, after installing it i don't see any pods running

**Comment by fahimistiaq91** [comment]
Can any tell what is correct procedure ?

**Comment by aneurinprice** [comment]
Hi, apologies for the delay.

When you install  helm, what logging do you see? Having some logs here would really help.

**Comment by zyyw** [comment]
Tried deploying a Harbor via harbor-helm (v1.12.2, harbor oss: v2.8.2) exposed through `ingress` with custom certificate (valid for 10 years), it succeeded.
1. create a secrete with custom certificate
```
kubectl create secret tls ingress-tls-secret-1 --cert=/home/ubuntu/certs/leaf.pem --key=/home/ubuntu/certs/leaf.key
```
2. deploy harbor-helm 
```
helm install harbor-ingress-20230615 --set expose.tls.certSource=secret,expose.tls.secret.secretName=ingress-tls-secret-1,persistence.enabled=false,externalURL=https://harbor.1.2.3.4.nip.io,expose.ingress.hosts.core=harbor.1.2.3.4.nip.io,imagePullSecrets[0].name=dockerhub-cred  .
```
3. pods are running; harbor portal can be accessed; verified that the validity of the deployed harbor is 10 years

**Comment by zyyw** [comment]
Tried deploying a Harbor via harbor-helm (v1.12.2, harbor oss: v2.8.2) exposed through `nodePort` with custom certificate (valid for 10 years), it succeeded.
1. create a secrete with custom certificate
```
kubectl create secret tls nodeport-tls-secret-1 --cert=/home/ubuntu/certs/nodeport/leaf.pem --key=/home/ubuntu/certs/nodeport/leaf.key 
```
2. deploy harbor-helm 
```
helm install harbor-nodeport-20230615 --set expose.tls.certSource=secret,expose.tls.secret.secretName=nodeport-tls-secret-1,persistence.enabled=false,externalURL=https://1.2.3.4:30003,expose.type=nodePort,expose.tls.auto.commonName=1.2.3.4,imagePullSecrets[0].name=dockerhub-cred .
```
3. pods are running; harbor portal can be accessed; verified that the validity of the deployed harbor is 10 years
